sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleIterator.scala

```diff
@@ -194,21 +265,26 @@ class RapidsShuffleIterator(
       s"${clients.size} clients.")
   }
 
-  taskContext.addTaskCompletionListener[Unit] { _ =>
-    //TODO: on task completion we currently don't ask clients to stop/clean resources
+  private[this] def receiveBufferCleaner(): Unit = {
     if (hasNext) {
       logWarning(s"Iterator for task ${taskContext.taskAttemptId()} closing, " +
-        s"but it is not done. Closing ${resolvedBatches.size()} resolved batches!!")
-      resolvedBatches.forEach(id => {
-        GpuShuffleEnv.getReceivedCatalog.removeBuffer(id)
-      })
+          s"but it is not done. Closing ${resolvedBatches.size()} resolved batches!!")
+      resolvedBatches.forEach {
+        case BufferReceived(bufferId) =>
+          GpuShuffleEnv.getReceivedCatalog.removeBuffer(bufferId)
+        case _ =>
+      }
     }
   }
 
+  //TODO: on task completion we currently don't ask clients to stop/clean resources
+  taskContext.addTaskCompletionListener[Unit] { _ =>
```

Comment:
Rather than passing a task context explicitly for unit tests and needing to mock one up in the unit tests, it's simpler to do this instead:
```scala
Option(TaskContext.get()).foreach(_.addTaskCompletionListener[Unit](_ => receiveBufferCleaner()))
```